### ISSUE TITLE
Add stylelint-at-rule-no-children to list of plugins

### ISSUE THREAD

**Issue by adityavm** [stylelint/stylelint#2568]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

New rule where block declarations are disallowed inside at rules (mostly conditionals and nested).

---

**Examples**

Valid CSS  
```css
.foo {
  color: red;

  @media screen and (max-width: 480px) {
    color: blue;
  }
}
```

Invalid CSS  
```css
.foo {
  color: red;
}

@media screen and (max-width: 480px) {
  .foo {
    color: blue;
  }
}
```

**Reasoning**

This is to avoid:

1. Whole selectors declared inside conditional at rules.
2. Having multiple unnecessary declarations of the same selector in different parts of the file.

Long story short, this is a way to consolidate all declarations for a selector, so that a anyone looking through a file can be confident that their first hit for a selector is the only declaration they should be concerned with.

(Sidenote: This still allows multiple declarations outside of conditionals, because we want to allow overriding styles when the situation calls for it. We have been using this guideline within my team at work and it's made a lot of the CSS easier to understand between multiple devs.)

**Code**
I have the code + tests ready [adityavm/stylelint-at-rule-no-children][1].

[1]: https://github.com/adityavm/stylelint-at-rule-no-children

**Comment by adityavm** [comment]
I wasn't sure what the protocol is for requesting new features so I opened an issue rather than just make a blind PR.

**Comment by jeddy3** [comment]
> I wasn't sure what the protocol is for requesting new features so I opened an issue rather than just make a blind PR.

That [is the correct](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#creating-a-new-rule) protocol :)

> I have the code + tests ready adityavm/stylelint-at-rule-no-children.

That's an interesting rule. I think you've got it right from the get-go by creating this as a plugin!

Would you like to submit a PR that adds it [to the list](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/plugins.md) of plugins?

**Comment by adityavm** [comment]
@jeddy3 done. Too bad the PR number is one more than the issue number 😞

**Comment by davidtheclark** [comment]
Done in #2569.
